simtools/corsika/corsika_output.py

```diff
@@ -0,0 +1,1301 @@
+        file_name = Path(file_name).with_suffix(".yml")
+        output_config_path = self.io_handler.get_output_directory(self.label, "corsika")
+        output_config_file = output_config_path.joinpath(file_name)
+        save_dict_to_file(self.hist_config, output_config_file)
+
+    def _create_histogram_default_config(self):
+        """
+        Create a dictionary with the configuration necessary to create the histograms. It is used
+        only in case the configuration is not provided in a yaml file or dict.
+
+        Three histograms are created: hist_position with 3 dimensions (x, y positions and the
+        wavelength), hist_direction with 2 dimensions (directive cosinus in x and y directions),
+        hist_time_altitude with 2 dimensions (time and altitude of emission).
+
+        Four arguments are passed to each dimension in the dictionary:
+
+        "bins": the number of bins,
+        "start": the first element of the histogram,
+        "stop": the last element of the histogram, and
+        "scale" to define the scale of the bins which can be "linear" or "log". If "log",
+            the start and stop values have to be valid, i.e., >0.
+
+        Returns
+        -------
+        dict:
+            Dictionary with the configuration parameters to create the histograms.
+        """
+
+        if self.individual_telescopes is False:
+            xy_maximum = 1000 * u.m
+            xy_bin = 100
+
+        else:
+            xy_maximum = 16 * u.m
+            xy_bin = 64
+
+        histogram_config = {
+            "hist_position": {
+                "x axis": {
+                    "bins": xy_bin,
+                    "start": -xy_maximum,
+                    "stop": xy_maximum,
+                    "scale": "linear",
+                },
+                "y axis": {
+                    "bins": xy_bin,
+                    "start": -xy_maximum,
+                    "stop": xy_maximum,
+                    "scale": "linear",
+                },
+                "z axis": {
+                    "bins": 80,
+                    "start": 200 * u.nm,
+                    "stop": 1000 * u.nm,
+                    "scale": "linear",
+                },
+            },
+            "hist_direction": {
+                "x axis": {
+                    "bins": 100,
+                    "start": -1,
+                    "stop": 1,
+                    "scale": "linear",
+                },
+                "y axis": {
+                    "bins": 100,
+                    "start": -1,
+                    "stop": 1,
+                    "scale": "linear",
+                },
+            },
+            "hist_time_altitude": {
+                "x axis": {
+                    "bins": 100,
+                    "start": -2000 * u.ns,
+                    "stop": 2000 * u.ns,
+                    "scale": "linear",
+                },
+                "y axis": {"bins": 100, "start": 50 * u.km, "stop": 0 * u.km, "scale": "linear"},
```

Comment:
Why 50 km and not more? CORSIKA sims start earlier (think at 118 km) and we might miss here Cherenkov photons from the primary particle before interaction (for cosmic rays).

Reply:
For the example file I used 50 km was fine but you are right, the default value should deal with all the cases. I changed it to 120 km now.